service/project.go

```diff
@@ -2,32 +2,45 @@ package service
 
 import (
 	"context"
+	"fmt"
 
 	"release-manager/pkg/apierrors"
 	"release-manager/pkg/dberrors"
+	"release-manager/pkg/githuberrors"
 	"release-manager/service/model"
 
 	"github.com/google/uuid"
 )
 
+const (
+	// GithubRepositoryTagsLimit is the maximum number of tags to fetch from a GitHub repository.
+	githubRepositoryTagsLimit = 20
+)
+
 type ProjectService struct {
-	authSvc        model.AuthService
-	projectRepo    model.ProjectRepository
-	envRepo        model.EnvironmentRepository
-	invitationRepo model.ProjectInvitationRepository
+	authSvc             model.AuthService
+	settingsSvc         model.SettingsService
+	projectRepo         model.ProjectRepository
+	envRepo             model.EnvironmentRepository
+	invitationRepo      model.ProjectInvitationRepository
+	githubClientFactory model.GithubClientFactory
```

Comment:
A GitHub client is created in the service only if all conditions are met (integration enabled, project has repository...) Therefore, the service accepts a `GitHubClientFactory` interface.

Reply:
That's a valid point; I think it deserves a comment.

Reply:
Or consider simply having `githubClient` as a pointer.

Reply:
(to get rid of the factory which feels overly complex)

Reply:
Thanks, it was actually very easy to implement it without the factory.